cmd/deletefailed.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright © 2021 NAME HERE <EMAIL ADDRESS>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"context"
+	"sync"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// deletefailedCmd represents the deletefailed command
+var Deletefailed = &cobra.Command{
+	Use:   "deletefailed",
+	Short: "Delete the non running Pods",
+	Run: func(cmd *cobra.Command, args []string) {
+		p := getPlatform(cmd)
+		client, err := p.GetClientset()
+		if err != nil {
+			p.Fatalf("unable to get clientset: %v", err)
+		}
+
+		// gather the list of Pods from all
+		pods, err := client.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			p.Fatalf("Failed to gather the list of jobs from namespace %v: %v", namespace, err)
+		}
+
+		// WaitGroup to synchronize go routines execution
+		wg := sync.WaitGroup{}
+
+		for _, po := range pods.Items {
+			wg.Add(1)
+
+			go func(po v1.Pod, clientSet *kubernetes.Clientset) {
+
+				condition := po.Status.Phase
+
+				// if the Condition of the Pod is not running, delete the Pod
+				if condition != "Running" {
```

Comment:
I believe this step can be done by setting  `ListOptions.FieldSelector` to `"status.Phase!=Running"` in the initial pod list retrieval.  FieldSelector has a fairly limited set of filterable fields, pod it looks like pods support status.phase: https://github.com/flant/shell-operator/blob/8fa3c3b8cfeb1ddb37b070b7a871561fdffe788b/HOOKS.md#fieldselector

Reply:
@BrendanGalloway I tried to incorporate all the PR suggestions, Can you please review now ? and kindly suggest if any other changes are required. 
![image](https://user-images.githubusercontent.com/36641301/123688899-39a54000-d870-11eb-8550-60d57d141574.png)